docs/guides/nextjs.md

```diff
@@ -99,14 +139,16 @@ export const CounterStoreProvider = ({
   )
 }
 
-export const useCounterStore = (selector: (store: CounterStore) => T): T => {
+export const useCounterStore = <T = unknown,>(
+  selector: (store: CounterStore) => T,
+): T => {
   const counterStoreContext = useContext(CounterStoreContext)
 
-  if (counterStoreContext === undefined) {
+  if (!counterStoreContext) {
     throw new Error(`useCounterStore must be use within CounterStoreProvider`)
   }
 
-  return useStore(counterStoreContext, selector)
+  return useStoreWithEqualityFn(counterStoreContext, selector)
```

Comment:
On second look, I'm not sure why we need `WithEqualityFn` here. Can you double check?

Reply:
@dbritto-dev well, the deprecation message said that we should use `useStoreWithEqualityFn`

Reply:
We are hoping to show only if it has 3 arguments:
https://github.com/pmndrs/zustand/blob/eb8443faa333e5804b372b211424eec05e200b47/src/react.ts#L43-L51

Otherwise, it's a false positive. If that's the case, I see why people reported about the deprecation message more than expected.

Reply:
@dai-shi I'm getting that deprecation message even with 2 arguments

Reply:
![image](https://github.com/pmndrs/zustand/assets/1697714/15d37e4f-6725-4ce3-b86f-fee488779563)

Reply:
Oh, no. So, maybe with just one argument?

Reply:
<img width="897" alt="image" src="https://github.com/pmndrs/zustand/assets/1697714/6b19b81f-4125-43b4-af19-d493d4993899">

Reply:
@dai-shi  same

Reply:
It works fine with TypeScript playground: https://tsplay.dev/Na5YBN

<img width="432" alt="image" src="https://github.com/pmndrs/zustand/assets/490574/a25a1067-1ea8-435a-a794-8ec03f5728e8">

Reply:
So, it doesn't actually show the deprecation message for your use case.

Reply:
https://github.com/pmndrs/zustand/pull/2337

Reply:
LGTM